danger-klaxit/lib/danger_plugin.rb

```diff
@@ -272,4 +283,61 @@ def public_methods_for_class_or_module(ast_class, class_parents = [])
       end
     end
   end
+
+  # Read the config.yml file (in a very custom way)
+  #
+  # @return [Hash]
+  def config_file
+    config = {}
+    name_regex = /^(?<name>[a-z]+):/i
+    var_regex = /^  (?<line>[a-z_]+:.+)$/
+    block = nil
+    File.readlines("config/config.yml").each do |line|
+      if (name_matches = line.match(name_regex))
+        block = name_matches[:name]
+        config[block] = []
+        next
+      end
+      if (var_matches = line.match(var_regex))
+        config[block] << var_matches[:line]
+      end
+    end
+    config
+  end
+
+  # Create a diff based on the config and its sorted version
+  #
+  # @param config [Hash]
+  # @return [Hash]
+  def diff_for_sorted_config(config)
+    result = {}
+    config.each do |(block, lines)|
+      sorted_lines = lines.sort
+      next if sorted_lines == lines
+      lines.each_with_index do |line, index|
+        if line != sorted_lines[index]
+          result[block] = [] if result[block].nil?
+          result[block] << { remove: line, add: sorted_lines[index] }
+        end
+      end
+    end
+    result
+  end
+
+  # Format diff for displaying
+  #
+  # @param [Hash] diff
+  # @return [String]
+  def formatted_diff(structured_diff)
```

Comment:
The _diff_ you output is really not a usual _diff_. This is a source of misunderstanding. Why not just print out the correct file, so user only has to copy/paste ? And this will be much easier to maintain as well ;)

you could also use the `<details>` tag in GH Flavored markdown to have a simpler output:

```md
<details>
<summary> config/config.yml </summary>

```yml
DEFAULTS: &DEFAULTS
  b: b
  c: c
  a: a

test:
  <<: *DEFAULTS
  d: d
\```
</details>
```
<details>
<summary> config/config.yml </summary>

```yml
DEFAULTS: &DEFAULTS
  b: b
  c: c
  a: a

test:
  <<: *DEFAULTS
  d: d
```
</details>